semgrep-core/src/engine/Match_tainting_rules.ml

```diff
@@ -293,9 +304,10 @@ let check_rule rule match_hook (default_config, equivs) taint_spec xtarget =
                   match_hook str m.env m.tokens))
     |> List.map (fun m -> { m with PM.rule_id = convert_rule_id rule.Rule.id })
   in
-  {
-    RP.matches;
-    errors;
-    skipped_targets = [];
-    profiling = { RP.rule_id = fst rule.Rule.id; parse_time; match_time };
-  }
+  ( {
+      RP.matches;
+      errors;
+      skipped_targets = [];
+      profiling = { RP.rule_id = fst rule.Rule.id; parse_time; match_time };
+    },
+    debug_taint )
```

Comment:
It's a bit ugly to tack it on at the end like this. How bad would it be to add this as a field in the previous record? In the future we will probably also want to send debugging information for the normal patterns too (what debugging_json used to be) so it would be nice to have this set up well.

Reply:
Totally agree! In this PR I just did the work needed inside the dataflow engine.  There should be another PR that actually carries this info all the way to `semgrep-core`'s output which I think it's a better time to tackle this issue. I left a TODO note.